python/python/bystro/utils/tar.py

```diff
@@ -0,0 +1,77 @@
+"""Utility functions for safely invoking tar in cross-OS manner."""
+
+import subprocess
+from enum import Enum
+from typing import NoReturn
+
+
+class OperatingSystem(Enum):
+    """Represent the operating system on which the current Python process is running."""
+
+    linux = "linux"
+    macosx = "macosx"
+
+
+# here and throughout, fully qualify executable names to avoid privilege escalation
+GNU_TAR_LINUX = "/usr/bin/tar"
```

Comment:
Can we use something like `which` to locate this? I'm not sure what privilege escalation is possible with that approach, please help me understand

Reply:
Thanks, good call, updated to use which to locate these.  

In re privilege escalation, if command line names aren't fully qualified, a malicious user with access to $PATH could save an executable named `tar` in a location that had precedence over /usr/bin and get up to shenanigans.  Not the biggest risk in this scenario but it's a belt and suspenders kind of thing.